File: 1441 build-an-array-with-stack-operations/solution.cs
public class Solution {
    public IList<string> BuildArray(int[] target, int n) {
        IList<string> ans = new List<string>();
        int idx=0;
        for(int i=1;i<=n && idx<target.Length;i++) {
            ans.Add("Push");
            if(target[idx]==i){
                idx++;
            }
            else {
                ans.Add("Pop");
            }
        }
        return ans;
    }
}

File: 1441 build-an-array-with-stack-operations/solution.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut ans : Vec<String> = Vec::new(); 
        let mut idx=0;

        for i in 1..=n {
            if idx==target.len() {
                break;
            } 
            ans.push("Push".to_string());
            if target[idx]==i {
                idx+=1;
            } else {
                ans.push("Pop".to_string());
            }
        }
        ans
    }
}
